src/c_bindings/c_wrapper.cpp

```diff
@@ -7,40 +7,57 @@
 extern "C" {
     // C wrapper function
     const char* create_discriminant_wrapper(const uint8_t* seed, size_t seed_size, int length) {
-        std::vector<uint8_t> seedVector(seed, seed + seed_size);
-        integer result = CreateDiscriminant(seedVector, length);
+        try {
+            std::vector<uint8_t> seedVector(seed, seed + seed_size);
+            integer result = CreateDiscriminant(seedVector, length);
 
-        // Serialize the 'result' to a string
-        std::string resultStr = result.to_string();
+            // Serialize the 'result' to a string
+            std::string resultStr = result.to_string();
 
-        // Allocate a new C-string to hold the serialized result
-        char* cResultStr = new char[resultStr.length() + 1];
-        std::strcpy(cResultStr, resultStr.c_str());
+            // Allocate a new C-string to hold the serialized result
+            char* cResultStr = new char[resultStr.length() + 1];
+            std::strcpy(cResultStr, resultStr.c_str());
 
-        return cResultStr; // Return the C-string
+            return cResultStr; // Return the C-string
+        } catch (const runtime_error& e) {
+            return NULL;
+        }
     }
 
     ByteArray prove_wrapper(const uint8_t* challenge_hash, size_t challenge_size, const uint8_t* x_s, size_t x_s_size, int discriminant_size_bits, uint64_t num_iterations) {
-        std::vector<uint8_t> challenge_hash_bytes(challenge_hash, challenge_hash + challenge_size);
-        integer D = CreateDiscriminant(challenge_hash_bytes, discriminant_size_bits);
-        form x = DeserializeForm(D, (const uint8_t*)x_s, x_s_size);
-        std::vector<uint8_t> result = ProveSlow(D, x, num_iterations);
-
-        // Allocate memory for the result and copy data
-        uint8_t* resultData = new uint8_t[result.size()];
-        std::copy(result.begin(), result.end(), resultData);
-
-        // Create and return a ByteArray struct
-        ByteArray resultArray = { resultData, result.size() };
-        return resultArray;
+        try {
+            std::vector<uint8_t> challenge_hash_bytes(challenge_hash, challenge_hash + challenge_size);
+            integer D = CreateDiscriminant(challenge_hash_bytes, discriminant_size_bits);
+            form x = DeserializeForm(D, (const uint8_t*)x_s, x_s_size);
+            std::vector<uint8_t> result = ProveSlow(D, x, num_iterations);
+
+            // Allocate memory for the result and copy data
+            uint8_t* resultData = new uint8_t[result.size()];
+            std::copy(result.begin(), result.end(), resultData);
+
+            // Create and return a ByteArray struct
+            ByteArray resultArray = { resultData, result.size() };
+            return resultArray;
+        } catch (const runtime_error& e) {
+            ByteArray resultArray = { NULL, 0 };
```

Comment:
`NULL` is the `C89` name (which is implicitly convertible to other types), `nullptr` is only convertible to a pointer

Reply:
Updated this